libs/designer/src/lib/ui/panel/recommendation/recommendationPanelContext.tsx

```diff
@@ -94,21 +104,27 @@ export const RecommendationPanelContext = (props: RecommendationPanelContextProp
 
   const onOperationClick = useCallback(
     (id: string, apiId?: string) => {
-      const operation = (allOperations ?? []).find((o: DiscoveryOperation<DiscoveryResultTypes>) => {
-        return apiId ? o.id === id && o.properties?.api?.id === apiId : o.id === id;
+      const searchOperation = SearchService().getOperationById?.bind(SearchService());
+
+      const searchResultPromise = searchOperation
+        ? searchOperation(id)
+        : Promise.resolve((allOperations ?? []).find((o) => (apiId ? o.id === id && o.properties?.api?.id === apiId : o.id === id)));
+
+      searchResultPromise.then((operation) => {
+        if (!operation) return;
+        dispatch(selectOperationId(operation.id));
+        setSelectedOperation(operation);
```

Comment:
what is the need for this new line `setSelectedOperation(operation);`

Reply:
the selected operation used to be set synchronously (deleted line 52), now that it became async I needed to change that into a set method within an effect.